boot/systems.go

```diff
@@ -275,3 +278,48 @@ func observeSuccessfulSystems(model *asserts.Model, m *Modeenv) (*Modeenv, error
 	}
 	return m, nil
 }
+
+// InspectTryRecoverySystemOutcome obtains a tried recovery system status. When
+// no recovery system has been tried, the outcome will be
+// TryRecoverySystemOutcomeNoneTried. The caller is responsible for clearing the
+// bootenv once the status bas been properly acted on.
+func InspectTryRecoverySystemOutcome(dev Device) (outcome TryRecoverySystemOutcome, label string, err error) {
+	opts := &bootloader.Options{
+		// setup the recovery bootloader
+		Role: bootloader.RoleRecovery,
+	}
+	// TODO:UC20: seed may need to be switched to RW
+	bl, err := bootloader.Find(InitramfsUbuntuSeedDir, opts)
+	if err != nil {
+		return TryRecoverySystemOutcomeFailure, "", err
+	}
+
+	vars, err := bl.GetBootVars("try_recovery_system", "recovery_system_status")
+	if err != nil {
+		return TryRecoverySystemOutcomeFailure, "", err
+	}
+	status := vars["recovery_system_status"]
+	trySystem := vars["try_recovery_system"]
+
+	if status == "" && trySystem == "" {
+		return TryRecoverySystemOutcomeNoneTried, "", nil
+	}
+
+	// we expect both variables to be set
+	if status != "try" && status != "tried" {
+		return TryRecoverySystemOutcomeInconsistent, "", &errInconsistentRecoverySystemState{
+			why: fmt.Sprintf("unexpected recovery system status %q", status),
+		}
+	}
+	if trySystem == "" {
+		return TryRecoverySystemOutcomeInconsistent, "", &errInconsistentRecoverySystemState{
+			why: fmt.Sprintf("try recovery system is unset but status is %q", status),
+		}
+	}
```

Comment:
I think it would be easier to reason about all these conditions if they were in a switch something like this:

```go
switch {
    case status == "" && trySystem == "":
        // not trying a system
    case trySystem == "":
        // no system set, but we have a status set
    case status == "":
        // status is empty, but the system is set, implying we failed and got rebooted and the flag was cleared or we are in an otherwise inconsistent state
    case status != "try" && status != "tried":
        // unknown status variable
    case status == "tried":
        // success
    default:
         // something else happened (but what?)
}
```